### ISSUE TITLE
The loss function doesn't match the one mentioned in Sec 3 of paper ?

### ISSUE THREAD

**Issue by yw981** [ydkim1293/NLNL-Negative-Learning-for-Noisy-Labels#4]
Hello ,I'm very interested in your work and trying to reproduce your results.
Q1:
I found the loss function in (line 200,main_NL.py)
`
( (loss+loss_neg) / (float((labels>=0).sum())+float((labels_neg[:,0]>=0).sum())) ).backward()
`
It seems that it dosen't match (Eq.2) or any loss function mentioned in (Section 3)?

Q2:
'main_NL.py','main_PL.py','main_pseudo1.py','main_pseudo2.py'  are very similar to each other. Could you summarize the differeces and provide more comments or instructions of the code ?

Thanks a lot!

**Comment by houlinhe** [comment]
Did you find out the answer?
